lib/file-helper.js

```diff
@@ -45,6 +47,111 @@ exports.getNunjucksCode = path => {
   return content
 }
 
+// This helper function takes a path of a macro options file and
+// returns the options data grouped by tables to output in markup
+exports.getMacroOptions = componentName => {
+  let options = []
+  let processedOptions = []
+  let primaryTable = {
+    'name': 'Primary options',
+    'id': 'primary',
+    'options': []
+  }
+  processedOptions.push(primaryTable)
+
+  if (componentName === 'text-input') {
+    componentName = 'input'
+  }
+
+  try {
+    let optionsFilePath = path.join(paths.govukfrontendcomponents, componentName, 'macro-options.json')
+    options = JSON.parse(fs.readFileSync(optionsFilePath, 'utf8'))
+  } catch (err) {
+    console.error(err)
```

Comment:
Thanks @36degrees. We now `process.exit()` if any of these fail. We could also just `return` from the function but that wouldn't necessarily fail the build.

Reply:
Because we're not specifying an exit code when calling `process.exit()`, this will cause the build to exit with the success code, which means that Travis will just carry on as if the build worked.

I'd suggest either exiting with an error code, or just throwing a more readable error from here.

Reply:
Thanks @36degrees. Looking at [this](https://nodejs.org/api/process.html#process_process_exit_code), it looks like exiting with `1` would make sense?

Reply:
👍